src/main/c/truffleposix/truffleposix.c

```diff
@@ -36,11 +36,31 @@ SUCH DAMAGE.
 #include <dirent.h>
 #include <errno.h>
 #include <stdlib.h>
+#include <stdint.h>
 
 #include <sys/resource.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#include <sys/stat.h>
+
+struct truffleposix_stat {
+  uint64_t atime;
+  uint64_t mtime;
+  uint64_t ctime;
+  uint64_t nlink;
+  uint64_t rdev;
+  uint64_t blksize;
+  uint64_t blocks;
+  uint64_t dev;
+  uint64_t ino;
+  uint64_t size;
+  uint64_t mode;
+  uint64_t gid;
+  uint64_t uid;
```

Comment:
Is the idea here that you just picked a type large enough to deal with cross-platform differences in things like `gid_t` and `ino_t`?

Reply:
A big part of why I'm asking is the various `DefaultRubiniusConfiguration` subclasses define stat field sizes and offsets based on their native sizes. If we're going to skip past all that and use our own struct with our own offsets, it might make sense to either remove the other configuration values or update them.

Reply:
I think they're all big enough, and unsigned is ok for all of them. I wouldn't bet my life on it, but all specs pass. I will clear out old Rubinius config values later as I think most of them may now go.